tools/static_word_checks.py

```diff
@@ -0,0 +1,126 @@
+import argparse
+import os
+import sys
+
+
+class StaticCheck:
+    """Run simple checks on the entire document or specific lines."""
+
+    def __init__(self, replace: bool):
+        self.full_file_checks = [
+            # add more checks to the list as they are added
+        ]
+        self.line_checks = [
+            self.check_for_noreturn,
+            self.check_for_double_r,
+            self.check_for_double_u,
+            # add more checks to the list as they are added
+        ]
+        self.replace = replace
+
+    def check_for_noreturn(
+        self, filename: str, lineno: str, content: str
+    ) -> bool:
+        if "noreturn" in content.lower():
+            print(
+                "{filename}: Line {line_number} has phrase noreturn, "
+                "please edit and remove this.".format(
+                    filename=filename, line_number=lineno
+                )
+            )
+            return False
+        return True
+
+    def check_for_double_r(
+        self, filename: str, lineno: str, content: str
+    ) -> bool:
+        if "/r/" in content:
+            # checks for reddit.com urls
+            content = content.replace("/r/", "r/")
+            content = content.replace(
+                "reddit.comr/", "reddit.com/r/"
+            )  # handles reddit.com/r/... urls
+        if "/r/" in content:
+            if self.replace:
+                with open(filename, "w", encoding="utf-8") as fp:
```

Comment:
As mentioned before, let's not explicitly state encoding. I'd really like to see a case where something breaks because we're not explicit before using it.